### PR TITLE
Create session and send SMS Functionality

### PR DESCRIPTION
**Pull request by jenish4096**

### CODE REVIEW

File: BIDVerifyDocument.js [hunk 1]
```diff
@@ -94,6 +98,150 @@ const verifyDocument = async (dvcId, verifications, document) => {
   }
 }
 
+const sendSMS = async (communityInfo, smsTo, smsISDCode, smsTemplateB64, requestIdUuid) => {
```
Comment (manik-1kosmos) [inline BIDVerifyDocument.js:101]: sending SMS has nothing to do with document verification!

Comment (manik-1kosmos) [reply BIDVerifyDocument.js:101]: Move this to BIDMessaging

File: BIDVerifyDocument.js [hunk 2]
```diff
@@ -94,6 +98,150 @@ const verifyDocument = async (dvcId, verifications, document) => {
...
+    }
+
+    const templateText = smsTemplate
+      .toString()
+      .replace(/<tenantname>/, communityInfo.tenant.name)
+      .replace(/<link>/, api_response.url);
+
+    const smsTemplateB64 = Buffer.from(templateText).toString('base64');
+
+    await sendSMS(communityInfo, smsTo, smsISDCode, smsTemplateB64, requestIdUuid);
```
Comment (manik-1kosmos) [inline BIDVerifyDocument.js:236]: SMS can't be sent as part of this function

Comment (manik-1kosmos) [inline BIDVerifyDocument.js:236]: You are missing session polling

File: BIDMessaging.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+"use strict";
+const { v4: uuidv4 } = require('uuid');
+
+const BIDECDSA = require('./BIDECDSA');
+const BIDSDK = require('./BIDSDK');
+const fetch = require('node-fetch');
+
+const sendSMS = async (smsTo, smsISDCode, smsTemplate, sessionUrl) => {
+    try {
```
Comment (manik-1kosmos) [inline BIDMessaging.js:17]: sendSMS function has nothing to do with "sessinoUrl"
The caller is responsible for sending a pre-formatted text message which should be sent to downstream service as as a b64 string.

File: BIDMessaging.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        const encryptedRequestId = BIDECDSA.encrypt(JSON.stringify({
+            ts: Math.round(new Date().getTime() / 1000),
+            appid: 'fixme',
+            uuid: uuidv4()
+        }), sharedKey);
+
+        let headers = {
+            'Content-Type': 'application/json',
+            'charset': 'utf-8',
+            'X-TenantTag': tenantTag,
```
Comment (manik-1kosmos) [inline BIDMessaging.js:53]: why do you need tenantTag here?

File: BIDVerifyDocument.js [hunk 5]
```diff
@@ -94,6 +98,142 @@ const verifyDocument = async (dvcId, verifications, document) => {
...
+    if (api_response) {
+      api_response = await api_response.json();
+
+      if (api_response.data) {
+        let dec_data = BIDECDSA.decrypt(api_response.data, sharedKey);
+        api_response = JSON.parse(dec_data);
+      }
+    }
+
+    if (status !== 200) {
```
Comment (manik-1kosmos) [inline BIDVerifyDocument.js:217]: I don; think this manipulation should be done here. 
docuverify is already programmed to work properly.. return status/response as-is.

Comment (manik-1kosmos) [reply BIDVerifyDocument.js:217]: Did you check if you need any "decryption" here.

**Review by manik-1kosmos — CHANGES_REQUESTED** [review]
needs to be checked for completeness.

File: BIDMessaging.js (hunk 3 above)
Comment (jenish4096) [reply BIDMessaging.js:17]: okay got it

File: BIDMessaging.js (hunk 4 above)
Comment (jenish4096) [reply BIDMessaging.js:53]: i am calling admin console API so i think we should need tanantTag in the header.
i checked and i don't need to pass tenantTag in the header, i will remove it from the header

File: BIDVerifyDocument.js (hunk 5 above)
Comment (jenish4096) [reply BIDVerifyDocument.js:217]: i verified that we dont need decryption in error cases

**Pull request merged**